allocator.go

```diff
@@ -78,6 +78,80 @@ type Allocator struct {
 	opts options
 }
 
+type NodeInterface struct{
+	n Node
+	NodeMap map[NodeID]Node
+}
+
+type RangeInterface struct {
+	r        Range
+	RangeMap map[RangeID]Range
+}
+
+type Cluster interface {
+	ClusterWriter
+}
+
+type ClusterWriter interface{
+	AddNode()
+	RemoveNode () bool
+	AddRange ()
+	RemoveRange () bool
+}
+
+// AddNode Update or Add a new Node
+func(ni NodeInterface) AddNode() {
+	ni.NodeMap[ni.n.id] = ni.n
+}
```

Comment:
```suggestion
// AddNode Update or Add a new Node
func (n Nodes) AddNode(node Node) {
	n[node.id] = node
}
```

Reply:
We can do this as outlined above, the flow would be (as a user):

1. Caller calls `NewNode(...)`
2. Passes in result to this method.

We can also instead deprecate the `NewNode(...)` fn and instead make this method similar to that, so something like:

```suggestion
// AddNode Update or Add a new Node
func (n Nodes) AddNode(id NodeID, tags []string, resources map[Resource]int64) {
	n[node.id] = Node {
		id:        id,
		tags:      tags,
		resources: resources,
	}
}
```